component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java

```diff
@@ -561,13 +631,188 @@ protected void init(StreamDefinition outputStreamDefinition, OptionHolder option
      */
     @Override
     public void publish(Object payload, DynamicOptions dynamicOptions) {
-        if (!publisherURLOption.isStatic()) {
-            initClientConnector(dynamicOptions);
-        }
+        //get the dynamic parameter
         String headers = httpHeaderOption.getValue(dynamicOptions);
+        List<Header> headersList = HttpSinkUtil.getHeaders(headers);
+
+        if (authType.equals(HttpConstants.BASIC_AUTH) || authType.equals(HttpConstants.NO_AUTH)) {
+            sendRequest(payload, dynamicOptions, headersList);
+        } else {
+            sendOauthRequest(payload, dynamicOptions, headersList);
+        }
+    }
+
+    private void sendOauthRequest(Object payload, DynamicOptions dynamicOptions, List<Header> headersList) {
+        //generate encoded base64 auth for getting refresh token
+        String consumerKeyValue = consumerKey + ":" + consumerSecret;
+        String encodedRefreshAuth = "Basic " + encodeBase64(consumerKeyValue);
+        //check the availability of access token in the header
+        checkAccessToken(encodedRefreshAuth, dynamicOptions, headersList);
+        //send a request to API and get the response
+        int response = sendRequest(payload, dynamicOptions, headersList);
+        //if authentication fails then get the new access token
+        if (response == HttpConstants.AUTHENTICATION_FAIL_CODE) {
+            oauthFails(payload, dynamicOptions, headersList, encodedRefreshAuth);
+        } else if (response == HttpConstants.SUCCESS_CODE) {
+            log.info("Request send successfully.");
+        } else if (response == HttpConstants.INTERNAL_SERVER_FAIL_CODE) {
+            log.error(response + ": Internal server connection failure. Please pass the valid parameters. ");
+        } else {
+            log.error(response + ": Can not obtain access token.");
```

Comment:
We should say that we are dropping the message here. Then specify the cause. Please refer [1] for an example

[1] https://github.com/wso2-extensions/siddhi-io-tcp/blob/master/component/src/main/java/org/wso2/extension/siddhi/io/tcp/sink/TCPSink.java#L223